modules/tinymce/src/core/main/ts/keyboard/ArrowKeys.ts

```diff
@@ -32,7 +32,13 @@ const executeKeydownOverride = (editor: Editor, caret: Cell<Text>, evt: Keyboard
     { keyCode: VK.LEFT, ctrlKey: !os.isMacOS(), altKey: os.isMacOS(), action: MatchKeys.action(BoundarySelection.movePrevWord, editor, caret) },
     { keyCode: VK.UP, action: MatchKeys.action(ContentEndpointNavigation.moveV, editor, false) },
     { keyCode: VK.DOWN, action: MatchKeys.action(ContentEndpointNavigation.moveV, editor, true) }
-  ], evt).each((_) => {
+  ];
+  if (os.isMacOS()) {
+    keyPatterns.push({ keyCode: VK.UP, action: MatchKeys.action(CefNavigation.selectToEndPoint, editor, false), metaKey: true, shiftKey: true });
+    keyPatterns.push({ keyCode: VK.DOWN, action: MatchKeys.action(CefNavigation.selectToEndPoint, editor, true), metaKey: true, shiftKey: true });
+  }
```

Comment:
This is doing mutation which we generally try to avoid... this looks like it could be avoided by using a spread inside the original and would also make sure we can order it appropriately.

For example:
```ts
  ...(os.isMacOS() ? [
    { keyCode: VK.UP, action: MatchKeys.action(CefNavigation.selectToEndPoint, editor, false), metaKey: true, shiftKey: true },
    { keyCode: VK.DOWN, action: MatchKeys.action(CefNavigation.selectToEndPoint, editor, true), metaKey: true, shiftKey: true }
  ] : [])
```